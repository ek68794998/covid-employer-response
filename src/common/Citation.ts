import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
