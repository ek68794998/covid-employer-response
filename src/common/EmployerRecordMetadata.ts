import { EmployerRating } from "./EmployerRating";
import { EmployerRecordBase } from "./EmployerRecordBase";

export class EmployerRecordMetadata extends EmployerRecordBase {
	public readonly negativeCount: number;

	public readonly positiveCount: number;

	public readonly ratingValue: EmployerRating;

	public constructor(
		negativeCount: number,
		positiveCount: number,
		ratingValue: EmployerRating) {

		super();

		this.negativeCount = negativeCount;
		this.positiveCount = positiveCount;
		this.ratingValue = ratingValue;
	}

	public get rating(): EmployerRating {
		return this.ratingValue;
	}

	public static getTrendIcon(e: EmployerRecordMetadata): "trending_up" | "trending_flat" | "trending_down" {
		switch (e.rating) {
			case "good":
				return "trending_up";

			case "poor":
				return "trending_down";

			case "fair":
			default:
				return "trending_flat";
		}
	}
}
