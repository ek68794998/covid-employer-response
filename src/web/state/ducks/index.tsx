import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});