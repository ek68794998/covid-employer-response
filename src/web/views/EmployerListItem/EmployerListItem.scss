@import "../Colors.scss";
@import "../Dimensions.scss";

$scale-base: 16px;

$body-color: $outline-color;
$border-radius: 12px;
$flag-size: 20px;
$icon-size: ($scale-base * 2);
$padding-interior: 12px;
$padding-sides: 16px;

.EmployerListItem__Container {
	background: $background;
	border: 1px solid $body-color;
	border-radius: $border-radius;
	display: flex;
	padding: 0 $padding-sides;
	transition: 0.1s all;
	width: 100%;

	.EmployerLogo__Image {
		height: $icon-size;
		margin-right: $padding-interior;
		width: $icon-size;
	}
}

.EmployerListItem__Container--NotFound {
	color: rgba(0, 0, 0, 0.5);
}

.EmployerListItem__Container--Simple {
	align-items: center;
	cursor: pointer;
	font: 400 24px 'Baloo Chettan 2';
	height: 56px;

	.EmployerListItem__Name {
		flex: 1;
		overflow: hidden;
		padding-right: 6px;
		text-overflow: ellipsis;
		white-space: nowrap;
	}

	.EmployerRatingPill__Box {
		align-items: center;
		cursor: pointer;
		display: flex;

		.material-icons {
			margin-left: 8px;
		}
	}

	&:hover {
		&, & .EmployerRatingPill__Box {
			color: #fff !important;
		}

		&.EmployerListItem__Rating--fair {
			background: $indicator-neutral;
			border-color: $indicator-neutral;
		}
	
		&.EmployerListItem__Rating--good {
			background: $indicator-positive;
			border-color: $indicator-positive;
		}
	
		&.EmployerListItem__Rating--poor {
			background: $indicator-negative;
			border-color: $indicator-negative;
		}
	}
}

.EmployerListItem__Container--Detailed {
	box-shadow: 0 0 0 $card-shadow-color;
	flex-direction: column;
	font-size: 16px;
	height: 250px;
	padding-bottom: $padding-interior;
	padding-top: $padding-interior;

	&:hover {
		box-shadow: 0px 3px 6px $card-shadow-color;
		transform: translateY(-1px);

		.EmployerListItem__AggregateRatings {
			opacity: 1.0;
		}

		a.EmployerActionLinks__Link {
			opacity: 1.0;
		}
	}

	.EmployerListItem__Summary {
		cursor: pointer;
		flex: 1;
		font-size: 14px;
		line-height: 180%;
		overflow-y: hidden;
		position: relative;
	
		p {
			margin: 0;
		}
	
		.EmployerListItem__OverflowScreen {
			position: absolute;
			bottom: 0;
			left: 0;
			right: 0;
			height: 30px;
			background: linear-gradient(to top, $background, rgba(255, 255, 255, 0));
		}
	}
	
	.EmployerListItem__Actions {
		display: flex;
		font-family: 'Baloo Chettan 2';
		justify-content: flex-end;
		margin: 3px 0 0;
	
		a.EmployerListItem__ReadMore {
			align-items: center;
			display: flex;
			flex: 1;
			justify-content: flex-end;
			margin-right: 0;
	
			i {
				margin-left: 4px;
			}
		}
	}
}

.EmployerListItem__Title {
	align-items: center;
	display: flex;
	font-family: 'Baloo Chettan 2';

	h2 {
		flex: 1;
		font-weight: 400;
		margin: 0 4px;
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;

		a {
			color: inherit;
		}
	}

	.EmployerLogo__Image {
		margin-right: ($padding-interior / 2);
	}
}

.EmployerListItem__Subtitle {
	color: #757575;
	cursor: default;
	display: flex;
	font-size: 12px;
	margin: ($padding-interior / 2) 0 $padding-interior;

	> a, > span {
		align-items: center;
		display: flex;
		margin-right: ($padding-interior * 2);
		transition: 0.2s all;

		&:hover {
			color: #212121;
		}
	}

	a {
		color: inherit;
	}

	i {
		font-size: 18px;
		margin-right: 4px;
		vertical-align: bottom;
	}

	.EmployerListItem__AggregateRatings {
		align-items: center;
		display: flex;
		flex: 1;
		justify-content: flex-end;
		margin: 0;
		opacity: 0.5;
		transition: 0.2s opacity;

		i {
			font-size: 24px;
			margin-right: 2px;
		}

		span {
			align-items: center;
			cursor: default;
			display: flex;
			margin-right: 8px;

			&.EmployerListItem__GoodRatings {
				color: $indicator-positive;
			}

			&.EmployerListItem__PoorRatings {
				color: $indicator-negative;
			}
		}
	}
}

.EmployerListItem__Location img {
	background: #a1a1a1;
	border-radius: 4px;
	box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
	height: ($flag-size * 3 / 4);
	margin: 0 ($padding-interior + 4px) 0 ($padding-interior / 2);
	width: $flag-size;
}

@media (max-width: $body-reduced-width) {
	.EmployerListItem__Title {
		flex-wrap: wrap;
	}

	.EmployerListItem__Title--noShort {
		.EmployerListItem__Subtitle {
			> a, > span {
				display: block;
				margin-top: 8px;
			}
		}

		.EmployerListItem__Links i {
			margin: 8px 6px 0 0 !important;
		}
	
		.EmployerListItem__LinkBreak {
			flex-basis: 100%;
			height: 0;
		}
	}

	.EmployerRatingPill__Box .material-icons {
		display: none;
	}
}

@media (any-hover: none) {
	.EmployerListItem__Subtitle .EmployerListItem__AggregateRatings {
		opacity: 1.0 !important;
	}
}

