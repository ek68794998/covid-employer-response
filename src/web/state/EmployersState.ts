import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}