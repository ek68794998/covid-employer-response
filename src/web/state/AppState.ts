import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}