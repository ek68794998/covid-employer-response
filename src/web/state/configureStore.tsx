import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;