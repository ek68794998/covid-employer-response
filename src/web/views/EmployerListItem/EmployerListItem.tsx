import React from "react";
import { useSelector } from "react-redux";
import { RouteProps, useHistory } from "react-router-dom";

import { EmployerRecordMetadata } from "../../../common/EmployerRecordMetadata";
import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { AppState } from "../../state/AppState";
import { getEmployerMetadata } from "../../state/ducks/employers/selectors";
import { getStrings } from "../../state/ducks/localization/selectors";

import EmployerLogo from "../EmployerLogo/EmployerLogo";

import "./EmployerListItem.scss";

interface Props extends RouteProps {
	employerId: string;

	onClick?: () => void;
}

const EmployerListItem: React.FC<Props> = (props: Props): React.ReactElement | null => {
	const strings: LocalizedStrings = useSelector(getStrings);

	const { push } = useHistory();

	const { employerId, onClick } = props;

	const employer: EmployerRecordMetadata | undefined =
		useSelector((state: AppState) => getEmployerMetadata(state, employerId));

	if (!employer) {
		return null; // TODO
	}

	const onClickEvent: () => void = onClick || ((): void => push(`/employers/${employer.id}`));

	return (
		<div
			className={`EmployerListItem__Container EmployerListItem__Rating--${employer.rating}`}
			onClick={onClickEvent}
		>
			<EmployerLogo employer={employer} />
			{employer.name}
		</div>
	);
};

export default EmployerListItem;
