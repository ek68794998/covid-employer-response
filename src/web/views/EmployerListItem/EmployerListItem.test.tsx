import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync, mockComponent } from "../../../__tests__/TestUtils";
import { EmployerRecordMetadata } from "../../../common/EmployerRecordMetadata";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerListItem from "./EmployerListItem";

jest.mock(
	"../EmployerDetailsHeader/EmployerDetailsHeader",
	() => mockComponent("EmployerDetailsHeader"));

jest.mock(
	"../EmployerActionLinks/EmployerActionLinks",
	() => mockComponent("EmployerActionLinks"));

describe("<EmployerListItem />", () => {
	test("renders standard without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({
			employers: {
				itemsMetadata: {
					foo: new EmployerRecordMetadata(0, 0, "fair"),
				},
			},
			strings: await getPlocStringsAsync(),
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListItem
							employerId={"foo"}
							showDetails={false}
						/>
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});

	test("renders details without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({
			employers: {
				itemsMetadata: {
					foo: new EmployerRecordMetadata(0, 0, "fair"),
				},
			},
			strings: await getPlocStringsAsync(),
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListItem
							employerId={"foo"}
							showDetails={true}
						/>
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
