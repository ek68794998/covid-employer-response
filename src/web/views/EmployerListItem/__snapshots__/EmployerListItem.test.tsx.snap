// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListItem /> renders details without exploding 1`] = `
<div
  className="EmployerListItem__Container EmployerListItem__Container--Detailed"
>
  <div
    className="EmployerListItem__Title"
  >
    <h2>
      <a
        onClick={[Function]}
        title=""
      >
        
      </a>
    </h2>
    <span
      className="EmployerRatingPill__Box EmployerRatingPill__Box--fair "
      title="Tλè ϱèñèřáℓ řáƭïñϱ ôƒ ƭλè è₥ƥℓô¥èř."
    >
      Fáïř
      <i
        className="material-icons"
      >
        trending_flat
      </i>
    </span>
  </div>
  <div
    className="EmployerListItem__Subtitle"
  >
    <span
      className="EmployerListItem__AggregateRatings"
      title="Âϱϱřèϱáƭè çôúñƭƨ ôƒ ƭλè ϱôôδ áñδ βáδ áçƭïôñƨ áñ è₥ƥℓô¥èř λáƨ ƭáƙèñ."
    >
      <span
        className="EmployerListItem__GoodRatings"
      >
        <i
          className="material-icons"
        >
          add
        </i>
        0
      </span>
      <span
        className="EmployerListItem__PoorRatings"
      >
        <i
          className="material-icons"
        >
          remove
        </i>
        0
      </span>
    </span>
  </div>
  <div
    className="EmployerListItem__Summary"
    onClick={[Function]}
  >
    <div
      className="EmployerListItem__OverflowScreen"
    />
  </div>
  <div
    className="EmployerListItem__Actions"
  >
    [component: EmployerActionLinks employer=([object Object]) /]
    <a
      className="EmployerListItem__ReadMore"
      onClick={[Function]}
    >
      Rèáδ ₥ôřè
      <i
        className="material-icons"
      >
        fullscreen
      </i>
    </a>
  </div>
</div>
`;

exports[`<EmployerListItem /> renders standard without exploding 1`] = `
<div
  className="EmployerListItem__Container EmployerListItem__Container--Simple EmployerListItem__Rating--fair"
  onClick={[Function]}
>
  <span
    className="EmployerListItem__Name"
  >
    
  </span>
  <span
    className="EmployerRatingPill__Box EmployerRatingPill__Box--fair EmployerRatingPill__Box--LowContrast"
    title="Tλè ϱèñèřáℓ řáƭïñϱ ôƒ ƭλè è₥ƥℓô¥èř."
  >
    Fáïř
    <i
      className="material-icons"
    >
      trending_flat
    </i>
  </span>
</div>
`;
